src/smolagents/agents.py

```diff
@@ -14,16 +14,22 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import importlib.resources
+import importlib
 import inspect
+import json
+import os
 import re
+import tempfile
 import textwrap
 import time
-from collections import deque
+from collections import Counter, deque
```

Comment:
See my suggestion about duplicate detection.
```suggestion
from collections import deque
```